sql/upgrade/sl3.0.sql

```diff
@@ -461,12 +474,34 @@ UPDATE :slschema.vendor SET entity_id = (SELECT id FROM entity WHERE 'V-' || ven
 
 UPDATE :slschema.customer SET entity_id = coalesce((SELECT min(id) FROM entity WHERE 'C-' || customernumber = control_code), entity_id);
 
+--TODO: Do we need that for MC?
+INSERT INTO defaults(setting_key,value)
+    SELECT 'curr',array_to_string(array_agg(curr),':')
+    FROM :slschema.curr
+    WHERE rn=1;
+
+INSERT INTO currency(curr,description)
+    SELECT curr,curr
+    FROM  :slschema.curr;
+
+-- Make sure currency table is complete
+SELECT DISTINCT curr, curr
+FROM (
+          SELECT DISTINCT curr FROM :slschema.ar
+    UNION SELECT DISTINCT curr FROM :slschema.ap
+    UNION SELECT DISTINCT curr FROM :slschema.gl
+) xx
+WHERE NOT EXISTS (
+    SELECT 1 FROM :slschema.curr c
+    WHERE c.curr = xx.curr AND curr is not null
+);
+
 --Entity Credit Account
 
 UPDATE :slschema.vendor SET business_id = NULL WHERE business_id = 0;
 INSERT INTO entity_credit_account
 (entity_id, meta_number, business_id, creditlimit, ar_ap_account_id,
-        cash_account_id, startdate, enddate, threshold, entity_class)
+        cash_account_id, startdate, enddate, threshold, entity_class, curr)
```

Comment:
The "curr" here isn't new for mc, right? Just trying to understand what are fixes and what are incremental improvements.

Reply:
Currencies were not migrated in SL3.0 transactions, so all were appearing under the default currency.,

Reply:
Trigger insert into transaction table